src/execution.rs

```diff
@@ -396,7 +396,9 @@ impl Scan {
             for (index, arm) in self.arms.iter().enumerate() {
                 let captures = arm.regex.captures(&match_string[i..]);
                 if let Some(captures) = captures {
-                    matches.push((captures, index));
+                    if !captures.get(0).unwrap().range().is_empty() {
```

Comment:
Hmm... Is there a situation where you would want to have (potentially) an empty match (which will now be skipped), or would it perhaps make more sense to throw an error if an empty match is ever encountered?

I can't off the top of my head think of a situation where allowing empty matches is actually beneficial, but admittedly I haven't used `scan` in this way much.

Reply:
That is a good point. I don't think allowing an empty match is ever useful. If it was intended, you can just lift those statements out of the scan, as it is unconditional. Throwing an error helps pointing out the issue to the user, probably improving the overall experience.